collector/src/main/java/com/bitmovin/analytics/license/LicenseCall.java

```diff
@@ -15,54 +19,66 @@
 public class LicenseCall {
     private static final String TAG = "BitmovinAnalytics";
     private BitmovinAnalyticsConfig config;
+    private Context context;
     private HttpClient httpClient;
+    private LicenseProvider licenseProvider;
 
-    public LicenseCall(BitmovinAnalyticsConfig config) {
+    public LicenseCall(BitmovinAnalyticsConfig config, Context context, LicenseProvider licenseProvider) {
         this.config = config;
-        this.httpClient = new HttpClient(config.getContext(), BitmovinAnalyticsConfig.licenseUrl);
+        this.context = context;
+        this.licenseProvider = licenseProvider;
+        this.httpClient = new HttpClient(context, config.getLicenseUrl());
     }
 
-    public void authenticate(final LicenseCallback callback) {
-        LicenseCallData data = new LicenseCallData();
-        data.setKey(config.getKey());
+    private String getLicenseKeyForChecking(){
+        String key = config.getKey();
+        if (key == null || key.isEmpty()){
+            key = licenseProvider.getAnalyticsLicense();
+        }
+        return key;
+    }
+
+    public void authenticate(final OnLicenseValidated callback) {
+        final LicenseCallData data = new LicenseCallData();
```

Comment:
Not sure if that makes sense to have the `key` from the config here on the `LicenseCallData`. Seems like "re-using" a concept but introduces more complexity IMO. Maybe we can try to refactor this together in a short pairing session to see if we can make it more elegant?

Reply:
Hm... yeah... We could also put this into the LicenseProvider